core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -61,6 +86,16 @@ trait EntityComponent {
     type EntityQuery = Query[EntityTable, EntityRecord, Seq]
     type EntityAttributeQuery = Query[EntityAttributeTable, EntityAttributeRecord, Seq]
 
+    // understands how to translate EntityRecordLiteLifted into EntityRecord
+    implicit object EntityRecordLightShape
+      extends CaseClassShape(EntityRecordLiteLifted.tupled, EntityRecordBuilder.toEntityRecord)
+
+    // use as a replacement for entityQuery. This query never selects the all_attribute_values column and therefore
+    // will always return an EntityRecord where .allAttributeValues is None.
+    // anywhere this query is used, Slick needs the the EntityRecordLightShape object in implicit scope.
+    def withoutAllAttributeValues =
```

Comment:
showerthought: can we change the other path to be called `withAllAttributeValuesWhichWillMakeYourDatabaseSad` so we nudge people in the right direction?

Reply:
+1 for a nudge because we usually don't care about `allAttributeValues` - we don't even unmarshal it into the Entity!

I'd also like to go further: rename `EntityRecord` into a new `EntityRecordWithInlineAttributes` and remove `allAttributeValues` from the standard `EntityRecord`.  But I'm guessing that's a very intrusive change, so let's discuss that for a later story.

Reply:
I had the same thoughts - wanted to rename `entityQuery` to be obvious, but didn't want the regression risk of such a widespread change. Agree on doing this for a follow-on.